### ISSUE TITLE
sometimes cannot send message (only on iOS 8)

### ISSUE THREAD

**Issue by jukiginanjar** [MegaBits/SIOSocket#16]
Hi
I've experienced weird error and it happen in iOS8+ only, 
The error is SIOSocket intermittently didn't send the emit message to server.

Please help,

**Comment by iliu** [comment]
anislav, the reason this works i believe is because dispatch_sync might optimize this and just call the block on the same thread. the dispatch to main thread in the original code causes threading issues because the javascriptContext will run on 2 different threads. This is effectively the same solution that @jonathannorris had when he commented out the dispatch_async

**Comment by zulkis** [comment]
**illu**, In my opinion we must use exactly the same queue that JS context uses.
But unfortunately we cannot easily get it by dispatch_get_current_queue since it is deprecated...

Other approach is to create JSContext without WebView, but it is not so easy since socket.io js code uses many variables from it.

**Comment by iliu** [comment]
@zulkis , yup i saw your changes to use NSThread currentThread. I think it should work better than https://github.com/MegaBits/SIOSocket/pull/28 (see my comments in there)

**Comment by iliu** [comment]
@zulkis, to be safe you might also want to wrap around the evaluateScript call in - (void)on:(NSString *)event callback:(void (^)(SIOParameterArray *args))function just in case someone keeps a reference to socket and tries to register a callback on a different thread

**Comment by zulkis** [comment]
@illu, may be, but it is not the case.
But for me it seems weird to add callbacks in different place than creation block :)

**Comment by iliu** [comment]
@zulkis, fair point, but it's definitely possible. For example, if one writes an app that listens to different socket io events based upon a menu selection, it's possible that they might try to call `[self.socket on:@"dynamic event"]` directly in the callback block of the ui event, in which most likely it won't be on the web thread that UIWebView is running on.

**Comment by zulkis** [comment]
Yes, they might, it is true. I will implement it when I have free time for that.

**Comment by PengsreangSrun** [comment]
How can I get data back from server. I used this code but it does not work
[socket on: requestString callback: ^(SIOParameterArray *args) {
            NSString *response = [args firstObject];
           NSLog(@"dosad");
        }];
